src/traversing.js

```diff
@@ -168,10 +159,18 @@ jQuery.each({
 		return elem.contentDocument || jQuery.merge( [], elem.childNodes );
 	}
 }, function( name, fn ) {
+	var // Sorting (which implicitly de-dupes) is required for non-adjacent traversals
+		sort = name !== "prev" && name !== "next",
```

Comment:
Good point @markelog. I would guess that `.next()` and `.prev()` are rarely run on multiple elements. If that is really common, perhaps we can skip the `sort` variable and check for 'matched.length > 1` instead of `this.length > 1` below?